plugins/inputs/unbound/unbound.go

```diff
@@ -0,0 +1,163 @@
+// +build !windows
+
+package unbound
+
+import (
+	"bufio"
+	"bytes"
+	"fmt"
+	"os/exec"
+	"strconv"
+	"strings"
+	"time"
+
+	"github.com/influxdata/telegraf"
+	"github.com/influxdata/telegraf/filter"
+	"github.com/influxdata/telegraf/internal"
+	"github.com/influxdata/telegraf/plugins/inputs"
+)
+
+type runner func(cmdName string, UseSudo bool) (*bytes.Buffer, error)
+
+// Unbound is used to store configuration values
+type Unbound struct {
+	Stats   []string
+	Binary  string
+	UseSudo bool
+
+	filter filter.Filter
+	run    runner
+}
+
+var defaultStats = []string{"total.*", "num.*", "time.up", "mem.*"}
+var defaultBinary = "/usr/sbin/unbound-control"
+
+var sampleConfig = `
+  ## If running as a restricted user you can prepend sudo for additional access:
+  #use_sudo = false
+
+  ## The default location of the unbound-control binary can be overridden with:
+  binary = "/usr/sbin/unbound-control"
+
+  ## By default, telegraf gather stats for 3 metric points.
+  ## Setting stats will override the defaults shown below.
+  ## Glob matching can be used, ie, stats = ["total.*"]
+  ## stats may also be set to ["*"], which will collect all stats
+  stats = ["total.*", "num.*","time.up", "mem.*"]
+`
+
+func (s *Unbound) Description() string {
+	return "A plugin to collect stats from Unbound - a validating, recursive, and caching DNS resolver "
+}
+
+// SampleConfig displays configuration instructions
+func (s *Unbound) SampleConfig() string {
+	return sampleConfig
+}
+
+// Shell out to unbound_stat and return the output
+func unboundRunner(cmdName string, UseSudo bool) (*bytes.Buffer, error) {
+	cmdArgs := []string{"stats"}
+
+	cmd := exec.Command(cmdName, cmdArgs...)
+
+	if UseSudo {
+		cmdArgs = append([]string{cmdName}, cmdArgs...)
+		cmd = exec.Command("sudo", cmdArgs...)
+	}
+
+	var out bytes.Buffer
+	cmd.Stdout = &out
+	err := internal.RunTimeout(cmd, time.Millisecond*200)
+	if err != nil {
+		return &out, fmt.Errorf("error running unbound-control: %s", err)
+	}
+
+	return &out, nil
+}
+
+// Gather collects the configured stats from unbound_stat and adds them to the
+// Accumulator
+//
+// The prefix of each stat (eg MAIN, MEMPOOL, LCK, etc) will be used as a
```

Comment:
These aren't actual sections, and there is no unbound_stat.  I guess this is holdovers from an earlier version of the code, can you update it?